src/category_theory/abelian/transfer.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.preadditive.additive_functor
+import category_theory.abelian.basic
+import category_theory.limits.preserves.shapes.kernels
+import category_theory.adjunction.limits
+
+/-!
+# Transferring "abelian-ness" across a functor
+
+If `𝒜` is an additive category, `ℬ` is an abelian category,
+we have `a : 𝒜 ⥤ ℬ` `b : ℬ ⥤ 𝒜` (both preserving zero morphisms),
+`b` is left exact (that is, preserves finite limits),
+and further we have `adj : b ⊣ a` and `i : a ⋙ b ≅ 𝟭 𝒜`,
+then `𝒜` is also abelian.
+
+See https://stacks.math.columbia.edu/tag/03A3
+
+## Notes
+The hypotheses, following the statement from the Stack project,
+may appear suprising: we don't ask that the counit of the adjunction is an isomorphism,
+but just that we have some potentially unrelated isomorphism `i : a ⋙ b ≅ 𝟭 𝒜`.
+
+However Lemma A1.1.1 from [Elephant] shows that in this situation the counit itself
+must be an isomorphism, and thus that `𝒜` is a reflective subcategory of `ℬ`.
+
+Someone may like to formalize that lemma, and restate this theorem in terms of `reflective`.
+-/
+
+noncomputable theory
+
+namespace category_theory
+open category_theory.limits
+
+universes v u₁ u₂
+
+namespace abelian_of_adjunction
+
+variables {𝒜 : Type u₁} [category.{v} 𝒜] [preadditive 𝒜]
+variables {ℬ : Type u₂} [category.{v} ℬ] [abelian ℬ]
+variables (a : 𝒜 ⥤ ℬ)
+variables (b : ℬ ⥤ 𝒜) [functor.preserves_zero_morphisms b]
```

Comment:
These names are consistent with the Stacks project, but they are inconsistent with the rest of mathlib (unless I am mistaken and there is a very good reason to have these names in this context?). I would prefer if you named the functors something like `F` and `G`.

Reply:
Done!